components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/EndpointListenerLoader.java

```diff
@@ -108,11 +109,23 @@ public static void loadListeners() {
         }
     }
 
+    /**
+     * This loads both internal http and https apis.
+     */
     private static void loadInternalInboundApis() {
-        int internalInboundPort = HTTPEndpointManager.getInstance().getInternalInboundPort();
-        if (HTTPEndpointManager.getInstance().isAnyInternalApiEnabled()) {
-            HTTPEndpointManager.getInstance()
-                    .startListener(internalInboundPort, InboundHttpConstants.INTERNAL_INBOUND_ENDPOINT_NAME, null);
+
+        HTTPEndpointManager manager = HTTPEndpointManager.getInstance();
+
+        if (manager.isAnyInternalHttpApiEnabled()) {
+            HTTPEndpointManager.getInstance().startListener(manager.getInternalInboundHttpPort(),
+                    InboundHttpConstants.INTERNAL_HTTP_INBOUND_ENDPOINT_NAME, null);
+        }
+
+        if (ConfigurationLoader.isSslConfiguredSuccessfully() && manager.isAnyInternalHttpsApiEnabled()) {
```

Comment:
I think the 2 conditions need to be swapped since ConfigurationLoader.isSslConfiguredSuccessfully() means manager.isAnyInternalHttpsApiEnabled() is true and vice versa is not true.